bin/write_viral_gff.py

```diff
@@ -21,136 +20,213 @@ def get_ena_contig_mapping(ena_contig_file):
-                    if contigs[row["Contig"]]["end"] < row["End"]:
-                        contigs[row["Contig"]]["end"] = row["End"]
-
-                #   remove '.faa' trailing viphog ID
-                viphog = row["Best_hit"].strip(".faa")
-                direction = "-" if row["Direction"] == "-1" else "+"
-
-                #   change contig name if ena mapping required
-                if ena_mapping:
-                    contig_name = f'{ena_mapping[row["Contig"]]}-{row["Contig"]}'
+                    Contig=Contig.replace('|','')
+                    CDS_ID=CDS_ID.replace('|','')
+                    desc='phage_linear'
+
+                if Contig not in contig_protID.keys():
+                    contig_protID[Contig]=[CDS_ID]
+                    contig_data[Contig]=desc
                 else:
-                    contig_name = row["Contig"]
-
-                #   ID=ERZ2271866.1-NODE-1-length-21396-cov-5.122534;viphog=ViPhOG1;viphog_taxonomy=Phaeovirus
-                annotation = (
-                    f'ID={contig_name};viphog={viphog};viphog_taxonomy={row["Label"]}'
-                )
-                #   start with: ERZ2271866.1-NODE-1-length-21396-cov-5.122534	ViPhOG	proviral_region	1020	2050	.	-	.
-                gff.write(
-                    f'{contig_name}\t{viphog}\tviral_sequence\t{row["Start"]}\t{row["End"]}\t.\t{direction}'
-                    f"\t.\t{annotation}\n"
-                )
-    return contigs
-
-
-def write_metadata(
-    checkv_files, taxonomy_files, sample_prefix, virify_contigs, ena_mapping=None
+                    contig_protID[Contig].append(CDS_ID)
+
+                Direction=Direction.replace('-1','-').replace('1','+')
+
+                if not Best_hit == 'No hit':
+                    Best_hit=Best_hit.replace('.faa','')
+                    Label=line_l[7]
+                    annotation='viphog='+Best_hit+';'+'viphog_taxonomy='+Label
+                #else:
+                #    annotation='product=hypothetical_protein'
+
+                    values=(str(Start),str(End),Direction,annotation)
+                    prot_data[CDS_ID]=values
+
+    #return dictionaries
+    return contig_protID, prot_data, contig_data
+
+
+def write_gff(
+    checkv_files, taxonomy_files, sample_prefix, assembly_file, annot_dicts, ena_mapping=None
 ):
+
     if ena_mapping:
         ena_assembly_accession = list(ena_mapping.values())[0].split(".")[0]
-        output_filename = f"{ena_assembly_accession}_virify_contig_viewer_metadata.tsv"
+        output_filename = f"{ena_assembly_accession}_virify.gff"
     else:
-        output_filename = f"{sample_prefix}_virify_contig_viewer_metadata.tsv"
-    headers = (
-        "sequence_id\tcontig\tvirify_taxonomy\tstart_of_first_viphog\tend_of_last_viphog\tcheckv_provirus\t"
-        "checkv_quality\tmiuvig_quality\n"
-    )
-    checkv_dict, taxonomy_dict = {}, {}
+        output_filename = f"{sample_prefix}_virify.gff"
```

Comment:
Don't we need the `.tsv`? Sorry, I'm confused

Reply:
We don't need it anymore, checkv quality, taxonomy, and viphog annotation are now included in the attributes field of the gff output

Reply:
OK. We need to propagate this to any other pipelines that may be using this.
And we need to include this in the change log.